flexget/plugins/services/myepisodes.py

```diff
@@ -88,99 +86,154 @@ class MyEpisodes(object):
-                                        'your account data or see if the site is down.'), log)
+            # attempt authentication
+            self.http_session = self._login(config)
+
+        except plugin.PluginWarning as w:
+            log.warning(str(w))
+            return;
+
+        except plugin.PluginError as e:
+            log.error(str(e))
+            return
 
         for entry in task.accepted:
+            # mark the accepted entries as acquired
             try:
-                self.mark_episode(task, entry, opener)
+                self._validate_entry(entry)
+                entry['myepisodes_id'] = self._lookup_myepisodes_id(entry)
+                self._mark_episode_acquired(entry)
+
             except plugin.PluginWarning as w:
                 log.warning(str(w))
 
-    def lookup_myepisodes_id(self, entry, opener, session):
-        """Populates myepisodes_id field for an entry, and returns the id.
+    def _validate_entry(self, entry):
+        """
+        Checks an entry for all of the fields needed to comunicate with myepidoes
+        Return: boolean
+        """
+        if 'series_season' not in entry \
+                or 'series_episode' not in entry \
+                or 'series_name' not in entry:
 
-        Call will also set entry field `myepisode_id` if successful.
+            raise plugin.PluginWarning(
+                'Can\'t mark entry `%s` in myepisodes without series_season, series_episode and series_name '
+                'fields' % entry['title'], log)
 
-        Return:
-            myepisode id
+    def _lookup_myepisodes_id(self, entry):
+        """
+        Attempts to find the myepisodes id for the series
+        Return: myepisode id or None
+        """
 
-        Raises:
-            LookupError if entry does not have field series_name
+        # Do we already have the id?
+        myepisodes_id = entry.get('myepisodes_id')
+        if myepisodes_id:
+            return myepisodes_id
+
+        # have we previously recorded the id for this series?
+        myepisodes_id = self._retrieve_id_from_database(entry)
+        if myepisodes_id:
+            return myepisodes_id
+
+        # We don't know the id for this series, so it's time to search myepisodes.com for it
+        myepisodes_id = self._retrieve_id_from_website(entry)
+        if myepisodes_id:
+            return myepisodes_id
+
+        raise plugin.PluginWarning('Unable to determine the myepisodes id for: `%s`' % entry['title'], log)
+
+    def _retrieve_id_from_database(self, entry):
+        """
+        Attempts to find the myepisodes id in the database
+        Return: myepisode id or None
+        """
+        lc_series_name =  entry['series_name'].lower()
+        info = self.db_session.query(MyEpisodesInfo).filter(MyEpisodesInfo.series_name == lc_series_name).first()
+        if info:
+            return info.myepisodes_id
+        return None
```

Comment:
`return None` is redundant. It will return `None` if it exits the function without hitting a `return` statement.